### PR TITLE
Feature/exclude

### PR DESCRIPTION
**Pull request by luisesteves**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,11 +19,12 @@ Has we use a Continues Delivery, we need a way to create release notes automatic
...
-|---|---|---|
-| `to_revision` | End revision commit | false |
-| `from_revision` | From revision commit | false |
-| `message_regexp_filters` | Regexp filters like. ie '^MDM-\|^CTS:' | true |
+| Option | Description | Optional | Default | 
+|---|---|---|---|
+| `to_revision` | End revision commit | false | |
+| `from_revision` | From revision commit | false | |
+| `message_regexp_filters` | Regexp filters like. ie '^MDM-\|^CTS:' | true | `.` |
+| `remove_older_merge_commits` | Remove older commit's that already was merge | true | `false` |
```
Comment (portellaa) [inline README.md]: what do you think of the description: *Remove commit messages already merged* ?

File: lib/fastlane/plugin/logme/actions/logme.rb [hunk 2]
```diff
@@ -4,9 +4,20 @@ class LogmeAction < Action
...
-            .select{|message| message[regexp_filter] != nil}
+        changes = Actions.sh("git log --pretty=\"%s\" --no-merges #{options[:from_revision]}...#{options[:to_revision]}")
+                      .split(/\n/)
+
+        if options[:remove_older_merge_commits]
+          merges = Actions.sh("git log --pretty=\"%b\" --merges #{options[:from_revision]}...#{options[:to_revision]}")
+                       .split("* commit")
+
+          merges.delete_at(0) # empty value
+          merges.delete_at(0) # last merge
```
Comment (portellaa) [inline lib/fastlane/plugin/logme/actions/logme.rb:15]: 🤔 just to make sure that this is right. it seems really 🔨 😛

**Review by portellaa — APPROVED** [review]
I just left two comments, but this seems fine. 🍻

File: README.md (hunk 1 above)
Comment (luisesteves) [reply README.md]: Agree

File: lib/fastlane/plugin/logme/actions/logme.rb (hunk 2 above)
Comment (luisesteves) [reply lib/fastlane/plugin/logme/actions/logme.rb:15]: It seams, but isn't! The "* commit" is the first text that appear in the log. The split assumes that we have something in both sides. For example:
`"* commit hello".split("* commit")
 => ["", " hello"]`

**Review by portellaa — APPROVED** [review]

**Comment by p4checo** [comment]
I'll just leave this here since it was already merged.

We are basing these behavior on the assumption that merges' commit messages won't be customized. If by some reason they are, then the commit message may not contain the commits that the merge included, and this won't work properly.

In the same way, the commit messages may contain commits that somehow "break" that splitting pattern and the inferred commits could be wrong.

However, in the lack of a better option currently, this seems like a good compromise 👍

**Comment by portellaa** [comment]
I think we should create an issue with that problem to look at this later. There is a way to fix this, for sure.

What do you guys think? @p4checo @luisesteves

**Comment by p4checo** [comment]
I agree. There must be a better way! 💪😄

**Pull request merged**